lib/partitioninfo.coffee

```diff
@@ -46,15 +48,18 @@ getLogicalPartitions = (disk, offset, extendedPartitionOffset = offset, limit =
 getPartitions = (disk, offset, getLogical = true) ->
 	disk = Promise.promisifyAll(disk)
 	result = []
+	extended = null
 	readMbr(disk, offset)
 	.then (buf) ->
 		for p in getPartitionsFromMBRBuf(buf)
 			result.push(partitionDict(p, offset))
-			if MBR.Partition.isExtended(p.type) and getLogical
-				return getLogicalPartitions(disk, p.byteOffset())
-				.then (logicalPartitions) ->
-					result.push(logicalPartitions...)
-					result
+			if MBR.Partition.isExtended(p.type)
+				extended = p
```

Comment:
I'm not sure this is required. I don't know any tool that would create an image with more than one extended partition for testing.